com.github.camel-tooling.lsp.eclipse.client.tests.ui/tests/com.github.cameltooling.lsp.ui.tests/src/com/github/cameltooling/lsp/ui/tests/CamelRuntimeProviderFeatureTest.java

```diff
@@ -0,0 +1,260 @@
+import org.eclipse.reddeer.eclipse.core.resources.ProjectItem;
+import org.eclipse.reddeer.eclipse.ui.navigator.resources.ProjectExplorer;
+import org.eclipse.reddeer.eclipse.ui.views.log.LogView;
+import org.eclipse.reddeer.jface.text.contentassist.ContentAssistant;
+import org.eclipse.reddeer.junit.internal.runner.ParameterizedRequirementsRunnerFactory;
+import org.eclipse.reddeer.junit.runner.RedDeerSuite;
+import org.eclipse.reddeer.requirements.cleanerrorlog.CleanErrorLogRequirement;
+import org.eclipse.reddeer.requirements.cleanworkspace.CleanWorkspaceRequirement;
+import org.eclipse.reddeer.swt.impl.button.PushButton;
+import org.eclipse.reddeer.swt.impl.ctab.DefaultCTabItem;
+import org.eclipse.reddeer.workbench.handler.WorkbenchShellHandler;
+import org.eclipse.reddeer.workbench.impl.editor.DefaultEditor;
+import org.eclipse.reddeer.workbench.ui.dialogs.WorkbenchPreferenceDialog;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ErrorCollector;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+import org.junit.runners.Parameterized.UseParametersRunnerFactory;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import com.github.cameltooling.lsp.reddeer.editor.SourceEditor;
+import com.github.cameltooling.lsp.reddeer.preference.CamelRuntimeProvider;
+import com.github.cameltooling.lsp.reddeer.utils.CreateNewEmptyFile;
+import com.github.cameltooling.lsp.reddeer.utils.JavaProjectFactory;
+import com.github.cameltooling.lsp.ui.tests.utils.EditorManipulator;
+
+/*
+*
+* @author fpospisi
+*/
+@RunWith(RedDeerSuite.class)
+@UseParametersRunnerFactory(ParameterizedRequirementsRunnerFactory.class)
+public class CamelRuntimeProviderFeatureTest {
+
+	public static final String PROJECT_NAME = "catalog-feature-test";
+	public static final String CAMEL_CONTEXT = "camel-context.xml";
+	public static final String RESOURCES_CONTEXT_PATH = "resources/catalog-version-feature-context.xml";
+	public static final String SOURCE_TAB = "Source";
+
+	public static final String DEFAULT_PROVIDER = "Default";
+	public static final String SB_PROVIDER = "Spring Boot";
+	public static final String QUARKUS_PROVIDER = "Quarkus";
+	public static final String KARAF_PROVIDER = "Karaf";
+
+	public static final String KNATIVE = "knative";
+	public static final String MONGO = "mongo";
+	public static final String JMX = "jmx";
+
+	public static final String KNATIVE_PROP = "knative:type/typeId";
+	public static final String MONGO_PROP = "mongodb:connectionBean";
+	public static final String JMX_PROP = "jmx:serverURL";
+
+	private SourceEditor sourceEditor;
+	private int cursorPosition;
+
+	private String timePeriodfactor;
+	private static final String TIMEOUT_PERIOD_FACTOR_PROPETY_NAME = RedDeerProperties.TIME_PERIOD_FACTOR.getName();
```

Comment:
```suggestion
	private static final String TIMEOUT_PERIOD_FACTOR_PROPERTY_NAME = RedDeerProperties.TIME_PERIOD_FACTOR.getName();
```

Reply:
Thanks, actually this typo is in every test which is using this change, will create JIRA task for repair...

Reply:
I would suggest to extract this "period factor update stuff" (before/after methods, constants...) to some default class which can be reused in each UI test